### ISSUE TITLE
Convergence issues

### ISSUE THREAD

**Issue by TaWald** [YutingHe-list/GVSL#7]
Hello,

I am trying to include GVSL into our nnssl https://github.com/MIC-DKFZ/nnssl framework.
We have tried to run it on our brain MRI collection ([OpenMind](https://huggingface.co/datasets/AnonRes/OpenMind/tree/main)), but so far have always run into convergence issues.  
In our tests, the reconstruction loss learned reasonable features, however, the cross-correlation loss did not learn anything.

Have you ever tried training GVSL on brain MRI data?
Or do you have any ideas on what the most likely failure points may be?

Any help would be appreciated.

Best,
Tassilo

**Comment by YutingHe-list** [comment]
The similarity metric used in the loss function is not well-suited for MRI. Due to the high variability in MRI intensity distributions, intensity-based metrics like NCC (Normalized Cross-Correlation) often fail. It is recommended to consider alternative loss functions that are more robust to MRI-specific characteristics, such as mutual information, structural similarity (SSIM), or modality-invariant feature-based losses.

**Comment by TaWald** [comment]
Alright!
Thank you for that info!
